WooCommerce/Classes/ViewRelated/Dashboard/Settings/SettingsViewController.swift

```diff
@@ -162,10 +174,31 @@ private extension SettingsViewController {
     }
 
     func refreshViewContent() {
+        refreshCanCollectPayment()
         configureSections()
         tableView.reloadData()
     }
 
+    func refreshCanCollectPayment() {
+        /// Bail early if the card present payments feature is disabled
+        ///
+        guard ServiceLocator.featureFlagService.isFeatureFlagEnabled(.cardPresentPayments) else {
+            canCollectPayments = false
+            return
+        }
+
+        canCollectPayments = false
+
+        /// At the moment, we only expect one gateway account, but in the futue we may have multiple gateway accounts
+        /// each capable of supporting card present payments, so let's do this right now
+        ///
+        for gatewayAccount in gatewayAccounts {
```

Comment:
I am sure there is a way to do this via a combination of map and reduce. Or something like 

`canCollectPayments = gatewayAccounts.filter { $0.isCardPresentEligible == true}.isNotEmpty`

I am not sure that would make things much better though 😛

Reply:
There's [`Array.contains(where:_)`](https://developer.apple.com/documentation/swift/array/2297359-contains):

```swift
canCollectPayments = gatewayAccounts.contains(where: \.isCardPresentEligible)
```

Reply:
Can do! In https://github.com/woocommerce/woocommerce-ios/pull/4381/commits/775566b5dcc26310f255e389af0a617e923a744c I removed this change - I'm deferring view controller changes to the last pull in this series.